### PR TITLE
fix well model for gasoil thermal

### PR DESCRIPTION
**Pull request by totto82**
This PR contains two fixes
1) make sure the correct well  indices are used for gasoil+energy
2) fix 2 phase for WELLPI by making sure the fluidstate is accessed by the canonical phase index and not the active phase position provided by the phaseUsage.

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
I went for the simplest solution of using a mapping from the phase index from phaseUsage and the one used in opm-models. Synchronizing the indexes used by the wellState and opm-models should be a prioritized cleanup to avoid issues like this, IMO. @atgeirr Any chance for prioritizing the wellstate refactoring you started on some time ago?

File: opm/simulators/wells/StandardWell_impl.hpp [hunk 1]
```diff
@@ -4170,7 +4170,8 @@ namespace Opm
             // Note: E100's notion of PI value phase mobility includes
             // the reciprocal FVF.
             const auto connMob =
-                mobility[ flowPhaseToEbosCompIdx(p) ].value() * fs.invB(p).value();
+                mobility[ flowPhaseToEbosCompIdx(p) ].value()
+                    * fs.invB(flowPhaseToEbosPhaseIdx(p)).value();
```
Comment (bska) [inline opm/simulators/wells/StandardWell_impl.hpp:4174]: Cool.  We probably need the same fix in the [MultisegmentWell](https://github.com/OPM/opm-simulators/blob/9552d0cc8a62f243fe47c0b0879224b47f010835/opm/simulators/wells/MultisegmentWell_impl.hpp#L4337-L4339) implementation too.

Comment (totto82) [reply opm/simulators/wells/StandardWell_impl.hpp:4174]: Thanks for spotting, I will update with a fix to the MSW.

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
The failed test runs fine locally. I will try again.

**Comment by totto82** [comment]
jenkins build this please

**Comment by bska** [comment]
> The failed test runs fine locally. I will try again.

I've seen that failure in other contexts recently.  You **may** want to try incorporating #3024 too.

**Comment by totto82** [comment]
> I've seen that failure in other contexts recently. You may want to try incorporating #3024 too.
We should get in #3024 in asap.

**Comment by bska** [comment]
> > I've seen that failure in other contexts recently. You may want to try incorporating #3024 too.
> 
> We should get in #3024 in asap.

Kind of.  That PR definitely makes the test go through, but I don't know if I'm just papering over some underlying issue since I don't know the aquifer implementation well enough.

**Comment by bska** [comment]
I think this looks good, but maybe @GitPaean would like to have a look before we merge this?

**Comment by GitPaean** [comment]
> I think this looks good, but maybe @GitPaean would like to have a look before we merge this?

It looks good to me.  Especially the more clear way to handle `numWellEq`.

**Comment by bska** [comment]
> It looks good to me. Especially the more clear way to handle `numWellEq`.

Thanks a lot for looking at this.  Then we merge this.

**Pull request merged**